code/modules/clothing/clothing.dm

```diff
@@ -63,6 +63,8 @@
 		LoadComponent(pocket_storage_component_path)
 	if(can_be_bloody && ((body_parts_covered & FEET) || (flags_inv & HIDESHOES)))
 		LoadComponent(/datum/component/bloodysoles)
+	if(!icon_state)
+		item_flags |= ABSTRACT
```

Comment:
Which cases does this cover?
Will this help catch things that should have icon_state but don't?

Reply:
Quite the opposite, this catches things that don't have icon_states and shouldn't, namely the higher parent types of clothing that don't come with a defined icon_state because their children are the only ones that need an icon state, so things like `/obj/item/clothing/glasses`, `/obj/item/clothing/suit`, etc., as those shouldn't be spawning in the first place, and should be VERY obvious they shoudn't exist.

Reply:
This broke mob_holder because they are initialized without an icon_state